GoogleSignIn/Sources/GIDGoogleUser.m

```diff
@@ -153,10 +124,26 @@ - (void)updateAuthState:(OIDAuthState *)authState
     _authentication = [[GIDAuthentication alloc] initWithAuthState:authState];
     _profile = profileData;
     
-    // These three tokens will be generated in the getter and cached .
-    _cachedAccessToken = nil;
-    _cachedRefreshToken = nil;
-    _cachedIdToken = nil;
+    [self updateTokensWithAuthState:authState];
+  }
+}
+
+- (void)updateTokensWithAuthState:(OIDAuthState *)authState {
+  self.accessToken = [[GIDToken alloc] initWithTokenString:authState.lastTokenResponse.accessToken
```

Comment:
Before updating the value of these three properties, we should check that we're not about to set it to a new `GIDToken` instance that `isEqual` to the current value of the property.  The same would go for `nil` values for `idToken`.

Reply:
Done.